dvc/stage/run.py

```diff
@@ -144,14 +148,23 @@ def checkpoint_monitor(stage, callback_func):
         monitor_thread.join()
 
 
-def _checkpoint_run(stage, callback_func, done_cond):
+def _checkpoint_run(stage, callback_func, done_cond, proc):
     """Run callback_func whenever checkpoint signal file is present."""
     signal_path = os.path.join(stage.repo.tmp_dir, CHECKPOINT_SIGNAL_FILE)
     while True:
         if os.path.exists(signal_path):
-            _run_callback(stage, callback_func)
-            logger.debug("Remove checkpoint signal file")
-            os.remove(signal_path)
+            try:
+                _run_callback(stage, callback_func)
+            except Exception:  # pylint: disable=broad-except
```

Comment:
To handle exceptions that occur during checkpoint commit generation (outside of the main thread) we either need to catch general `Exception` here or override [threading.excepthook](https://docs.python.org/3/library/threading.html#threading.excepthook) with our own handler, but I think it's cleaner/easier to read this way